extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -29,43 +30,38 @@
 
     private static final Logger LOGGER = Logger.getLogger(CaffeineCacheImpl.class);
 
-    private AsyncCache<Object, Object> cache;
+    final AsyncCache<Object, Object> cache;
 
-    private String name;
+    private final CaffeineCacheInfo cacheInfo;
+    private final StatsCounter statsCounter;
 
-    private Integer initialCapacity;
-
-    private Long maximumSize;
-
-    private Duration expireAfterWrite;
-
-    private Duration expireAfterAccess;
-
-    public CaffeineCacheImpl(CaffeineCacheInfo cacheInfo) {
-        this.name = cacheInfo.name;
+    public CaffeineCacheImpl(CaffeineCacheInfo cacheInfo, boolean recordStats) {
+        this.cacheInfo = cacheInfo;
         Caffeine<Object, Object> builder = Caffeine.newBuilder();
         if (cacheInfo.initialCapacity != null) {
-            this.initialCapacity = cacheInfo.initialCapacity;
             builder.initialCapacity(cacheInfo.initialCapacity);
         }
         if (cacheInfo.maximumSize != null) {
-            this.maximumSize = cacheInfo.maximumSize;
             builder.maximumSize(cacheInfo.maximumSize);
         }
         if (cacheInfo.expireAfterWrite != null) {
-            this.expireAfterWrite = cacheInfo.expireAfterWrite;
             builder.expireAfterWrite(cacheInfo.expireAfterWrite);
         }
         if (cacheInfo.expireAfterAccess != null) {
-            this.expireAfterAccess = cacheInfo.expireAfterAccess;
             builder.expireAfterAccess(cacheInfo.expireAfterAccess);
         }
+        if (recordStats) {
+            statsCounter = new ConcurrentStatsCounter();
+            builder.recordStats(() -> statsCounter);
```

Comment:
Let's avoid lambdas in runtime code.

Reply:
Oops, I forgot about the no runtime lambda rule 😄